internal/services/databricks/databricks_virtual_network_peerings_resource.go

```diff
@@ -0,0 +1,326 @@
+	if err != nil {
+		return err
+	}
+
+	// for the import scenario, once the peering is created I can generate the workspace ID from the peering ID
+	// NOTE: this may not be right because I don't know if it is valid to create a peering with a
+	// workspace across subscription boundaries?
+	workspaceId := vnetpeering.NewWorkspaceID(id.SubscriptionId, id.ResourceGroupName, id.WorkspaceName)
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on Databricks %s: %+v", *id, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroupName)
+	d.Set("name", id.VirtualNetworkPeeringName)
+	d.Set("workspace_id", workspaceId.ID())
+
+	if model := resp.Model; model != nil {
+		d.Set("allow_virtual_network_access", model.Properties.AllowVirtualNetworkAccess)
+		d.Set("allow_forwarded_traffic", model.Properties.AllowForwardedTraffic)
+		d.Set("allow_gateway_transit", model.Properties.AllowGatewayTransit)
+		d.Set("use_remote_gateways", model.Properties.UseRemoteGateways)
+
+		if model.Properties.DatabricksAddressSpace != nil && model.Properties.DatabricksAddressSpace.AddressPrefixes != nil {
+			d.Set("address_space_prefixes", model.Properties.DatabricksAddressSpace.AddressPrefixes)
+		}
+
+		if model.Properties.RemoteAddressSpace != nil && model.Properties.RemoteAddressSpace.AddressPrefixes != nil {
+			d.Set("remote_address_space_prefixes", model.Properties.RemoteAddressSpace.AddressPrefixes)
+		}
+
+		if model.Properties.DatabricksVirtualNetwork != nil {
+			if databricksNetwork := model.Properties.DatabricksVirtualNetwork.Id; databricksNetwork != nil {
+				d.Set("virtual_network_id", databricksNetwork)
+			}
+		}
+
+		if remoteNetwork := model.Properties.RemoteVirtualNetwork.Id; remoteNetwork != nil {
+			d.Set("remote_virtual_network_id", remoteNetwork)
+		}
+	}
+
+	return nil
+}
+
+func resourceDatabricksVirtualNetworkPeeringDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataBricks.VnetPeeringsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// check to see if it exist first, this maybe a no op because when you delete a peering
+	// it auto deletes the corresponding peering in the other resource
+	existing, err := client.Get(ctx, *id)
+	if err != nil && response.WasNotFound(existing.HttpResponse) {
+		return nil
+	}
+
+	peerMutex.Lock(resourceLock)
+	defer peerMutex.Unlock(resourceLock)
+
+	// this is a workaround for the SDK issue: https://github.com/hashicorp/go-azure-sdk/issues/351
+	// and will be cleaned up once the fix has been implemented.
+	result, err := client.Delete(ctx, *id)
+	if err != nil && result.HttpResponse != nil {
+		return fmt.Errorf("deleting Databricks %s: %+v", *id, err)
+	}
```

Comment:
FYI this is now available as-of #21004 so we can make this:

```suggestion
	if err := client.DeleteThenPoll(ctx, *id); err != nil {
		return fmt.Errorf("deleting %s: %+v", *id, err)
	}
```

Reply:
Fixed.